File: src/FamilyOs.Application/Family/Dtos/FamilyMemberDto.cs
using FamilyOs.Domain.Enums;

namespace FamilyOs.Application.Family.Dtos;

public record FamilyMemberDto(
    Guid Id,
    string DisplayName,
    string? FullName,
    Relation Relation,
    DateOnly? BirthDate,
    string? Notes,
    bool HasUserAccount,
    string RowVersion,
    DateTime? DeletedUtc);

File: frontend/src/app/features/family/models/family-member.dto.ts
export type Relation = 'Self' | 'Spouse' | 'Child' | 'Parent' | 'Other';

export interface FamilyMemberDto {
  id: string;
  displayName: string;
  fullName: string | null;
  relation: Relation;
  birthDate: string | null;
  notes: string | null;
  hasUserAccount: boolean;
  rowVersion: string | null;
  deletedUtc: string | null;
}

export interface CreateFamilyMemberDto {
  displayName: string;
  fullName?: string;
  relation: Relation;
  birthDate?: string;
}

export interface UpdateFamilyMemberDto {
  displayName?: string;
  fullName?: string;
  relation?: Relation;
  birthDate?: string;
  rowVersion: string;
}
